src/mongocxx/collection.cpp

```diff
@@ -416,7 +416,8 @@ stdx::optional<result::replace_one> collection::replace_one(view_or_value filter
     class bulk_write bulk_op(bulk_opts);
 
     model::replace_one replace_op(filter, replacement);
-    if (options.upsert()) replace_op.upsert(options.upsert().value());
+    if (options.collation()) replace_op.collation(*options.collation());
+    if (options.upsert()) replace_op.upsert(*options.upsert());
```

Comment:
Almost all options applications in this file use a block form of `if`.  See https://gist.github.com/xdg/3870ef40ceabf1b9b0c9c592034ff740

I'd rather move towards that common style instead of repeating the uncommon style.

Why are we changing the upsert line?